### PR TITLE
Fix issue #58 cliOptions.fix is ignored

### PR DESCRIPTION
**Pull request by hassoubeat**
Hi !

I have fixed [issue 58](https://github.com/jest-community/jest-runner-eslint/issues/58).

I just got it to work for now, so if you have any comments, please comment!

### CODE REVIEW

File: src/runner/runESLint.js [hunk 1]
```diff
@@ -122,6 +122,15 @@ const getComputedFixValue = ({ fix, quiet, fixDryRun }) => {
   return undefined;
 };
 
+const removeUndefinedFromObject = object => {
+  return Object.keys(object).forEach(v => {
+    if (object[v] === undefined) {
+      // eslint-disable-next-line no-param-reassign
```
Comment (hassoubeat) [inline src/runner/runESLint.js:128]: The code using Object.assign timed out the test.
Please comment if you have any other good code.

File: src/runner/runESLint.js [hunk 2]
```diff
@@ -122,6 +122,15 @@ const getComputedFixValue = ({ fix, quiet, fixDryRun }) => {
...
+      delete object[v];
+    }
+  });
+};
```
Comment (ljharb) [inline src/runner/runESLint.js:132]: mutating `extraOptions` doesn't seem like a good idea
```suggestion
function removeUndefinedFromObject(object) {
  return Object.fromEntries(Object.entries(object).filter(([, value]) => typeof value !== 'undefined'));
};
```
and if the current supported node versions don't support Object.fromEntries and/or Object.entries, please use https://npmjs.com/object.fromentries and https://npmjs.com/object.entries.

**Review by ljharb — CHANGES_REQUESTED** [review]
This would need tests to cover whatever's being fixed.

File: src/runner/runESLint.js (hunk 2 above)
Comment (hassoubeat) [reply src/runner/runESLint.js]: Thanks for the comment!
I have incorporated your suggestion!

**Comment by ljharb** [comment]
Looks like you need to run prettier.

**Comment by hassoubeat** [comment]
@ljharb 

> This would need tests to cover whatever's being fixed.

Thanks for the comment.
I'd be glad to get your comments if I'm wrong in my plan on how to write test cases.

I am thinking that I could pass the [cliOptions](https://github.com/hassoubeat/jest-runner-eslint/blob/ad28de028a1bacb3621dc4af98bc1c6c0960ca5f/src/runner/runESLint.js#L168-L172) to the [mkTestResults](https://github.com/hassoubeat/jest-runner-eslint/blob/ad28de028a1bacb3621dc4af98bc1c6c0960ca5f/src/runner/runESLint.js#L51-L59) and make sure that the `cliOptions.fix` is not overwritten with undefined on the test code.

**Comment by ljharb** [comment]
That seems like it would work.

File: src/runner/__tests__/runESLint.test.js [hunk 3]
```diff
@@ -18,11 +22,16 @@ const runESLintRunnerWithMockedEngine = ({ mockOptions, runESLintOptions }) => {
       loadFormatter() {
         return Promise.resolve({ format() {} });
       }
+
+      // eslint-disable-next-line no-unused-vars
+      static outputFixes(report) {
+        return Promise.resolve();
+      }
```
Comment (ljharb) [inline src/runner/__tests__/runESLint.test.js:29]:
```suggestion
      async static outputFixes(report) {
      }
```

**Review by ljharb — APPROVED** [review]
LGTM assuming the test fails prior to the change, and passes after

**Review by SimenB — APPROVED** [review]
👍

**Pull request merged**